test/unit/connection_test.rb

```diff
@@ -111,13 +110,26 @@
   # =========================================================================== #
   #                         Running Hab CLI commands
   # =========================================================================== #
-  # TODO
-  # describe 'when cli non-transport options are passed' do
-  #   it 'should recognize them' do
-  #      # such as hab path
-  #      # such as env vars
-  #   end
-  # end
+
+  describe '#run_hab_cli' do
+    it 'should call the CLI connection while prefixing with the hab path' do
+      cli_cxn = mock
+      cli_cxn.expects(:run_command).with('/bin/hab testcmd')
+      conn.expects(:cli_options_provided?).returns(true)
+      conn.expects(:cli_connection).returns(cli_cxn)
+      conn.run_hab_cli('testcmd')
```

Comment:
I personally don't like 100% mock tests and I don't know this code base... Are there tests that verify that the above calls actually work correctly? (ie produce the correct side effects or return the correct result)

Reply:
Yes.  Please refer to the integration test.

Here, I do care that the expectations are met - specifically, the prefixing behavior on line 117.